File: homework01/caesar.py
def encrypt_caesar(plaintext):
    """
    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ""

    for letter in plaintext:
        if letter >= "A" and letter <= "Z":
            crypt_letterOrd = ord("A")
        elif letter >= "a" and letter <= "z":
            crypt_letterOrd = ord("a")
        else:
            ciphertext += letter
            continue

        crypt_letterOrd += (ord(letter) - crypt_letterOrd + 3) % 26
        ciphertext += chr(crypt_letterOrd)

    return ciphertext


def decrypt_caesar(ciphertext):
    """
    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ""

    for cryptLetter in ciphertext:
        if cryptLetter >= "A" and cryptLetter <= "Z":
            encrypt_letterOrd = ord("Z")
        elif cryptLetter >= "a" and cryptLetter <= "z":
            encrypt_letterOrd = ord("z")
        else:
            plaintext += cryptLetter
            continue

        encrypt_letterOrd -= (encrypt_letterOrd - ord(cryptLetter) + 3) % 26
        plaintext += chr(encrypt_letterOrd)

    return plaintext

File: homework01-go/caesar/caesar.go
package caesar

func EncryptCaesar(plaintext string, shift int) string {
	var ciphertext string
	var firstLetter int

	for ind := 0; ind < len(plaintext); ind++{
		if plaintext[ind] >= 'a' && plaintext[ind] <= 'z'{
			firstLetter = 'a'
		}else if plaintext[ind] >= 'A' && plaintext[ind] <= 'Z'{
			firstLetter = 'A'
		}else{
			ciphertext = ciphertext + string(plaintext[ind])
			continue
		}

		ciphertext = ciphertext + string(firstLetter + (int(plaintext[ind]) - firstLetter + 3) % 26)
	}

	return ciphertext
}

func DecryptCaesar(ciphertext string, shift int) string {
	var plaintext string
	var lastLetter int

	for ind := 0; ind < len(ciphertext); ind++{
		if ciphertext[ind] >= 'a' && ciphertext[ind] <= 'z'{
			lastLetter = 'z'
		}else if ciphertext[ind] >= 'A' && ciphertext[ind] <= 'Z'{
			lastLetter = 'Z'
		}else{
			plaintext = plaintext + string(ciphertext[ind])
			continue
		}

		plaintext = plaintext + string(lastLetter - (lastLetter - int(ciphertext[ind]) + 3) % 26)
	}

	return plaintext
}
